src/operator/mshadow_op.h

```diff
@@ -12,6 +12,13 @@
 namespace mxnet {
 namespace op {
 namespace mshadow_op {
+#ifdef __CUDA_ARCH__
+#define CONSTANT __constant__
+#else
+#define CONSTANT
```

Comment:
__constant__ makes it a actual variable, which is bad for performance.
I guess let's use #define MXNET_PI then.